src/pages/projectSettings/tabs/PluginsTab/PluginsTab.tsx

```diff
@@ -72,5 +73,21 @@ const validate: ValidateProps<FormState> = (formData, errors) => {
     );
   }
 
+  const displayNameDefined = !!patchMetadataPanelLink.displayName.trim();
+  const urlTemplateDefined = !!patchMetadataPanelLink.urlTemplate.trim();
+  if (displayNameDefined && !urlTemplateDefined) {
+    errors.externalLinks.patchMetadataPanelLink.urlTemplate.addError(
+      "You must specify a URL template or exclude display name."
+    );
+  } else if (!displayNameDefined && urlTemplateDefined) {
+    errors.externalLinks.patchMetadataPanelLink.displayName.addError(
+      "You must specify a display name or exclude URL template."
+    );
+  }
+  if (patchMetadataPanelLink.displayName.length > 40) {
```

Comment:
This can likely be handled by rjsf as well.
https://rjsf-team.github.io/react-jsonschema-form/docs/usage/dependencies/#schema-dependencies